test/server/lib/OIDCConfig.ts

```diff
@@ -241,6 +242,36 @@ describe('OIDCConfig', () => {
         });
       });
     });
+
+  describe('GRIST_HTTPS_PROXY', function () {
+    const proxyURL = 'http://localhost-proxy8080';
+    const httpAgent = new HttpProxyAgent(proxyURL);
+    [
+        {
+          itMsg: 'when omitted should not set proxyAgent to oidc-client',
+          expectedUserDefinedHttpOptions: { }
+        },
+        {
+          itMsg: 'should add proxyAgent to openid-client',
+          env: {
+            GRIST_HTTPS_PROXY: proxyURL,
+          },
+          expectedUserDefinedHttpOptions: {
+            agent: httpAgent
+          }
+        }
+      ].forEach(ctx => {
+        it(ctx.itMsg, async () => {
+          const setHttpOptionsDefaultsStub = sandbox.stub(custom, 'setHttpOptionsDefaults');
+          setEnvVars();
+          Object.assign(process.env, ctx.env);
+          const promise = OIDCConfigStubbed.buildWithStub();
+          await assert.isFulfilled(promise, 'initOIDC should have been fulfilled');
+          assert.isTrue(setHttpOptionsDefaultsStub.calledOnce, 'Should have called custom.setHttpOptionsDefaults');
+          Sinon.assert.match(setHttpOptionsDefaultsStub.firstCall.args[0], ctx.expectedUserDefinedHttpOptions);
```

Comment:
I don't see where we use `Sinon.assert`. My point is that it is a library the person behind the code should be aware of (and unfortunately, Sinon is a bit laconic and lazy with their documentation).

What about this instead?

```suggestion
          const actualProxy = _.omit(setHttpOptionsDefaultsStub.firstCall.args[0], 'agent.callback');
          assert.deepEqual(actualProxy, ctx.expectedUserDefinedHttpOptions);
```

Reply:
@jonathanperret pointed me out that we use `Sinon.assert` elsewhere. It is more convenient in our case for testing the calls to spies and stubs. Sorry for the bad suggestion above @tristanrobert!